src/main/java/nl/beehivejava/scriptj/parser/JsonPluginInformationParser.java

```diff
@@ -0,0 +1,148 @@
+package nl.beehivejava.scriptj.parser;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+import nl.beehivejava.scriptj.PluginInformation;
+import nl.beehivejava.scriptj.script.ScriptInformation;
+import nl.beehivejava.scriptj.script.ScriptType;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Can parse an {@link InputStream} to a {@link PluginInformation}, given that the {@code InputStream} contains its
+ * information in a JSON format.
+ *
+ * @author Lesley
+ */
+public final class JsonPluginInformationParser implements PluginInformationParser {
+
+    public static final String VALUE_ID = "id";
+    public static final String VALUE_AUTHOR = "author";
+
+    public static final String VALUE_SCRIPTS = "scripts";
+    public static final String VALUE_SCRIPT_NAME = "name";
+    public static final String VALUE_SCRIPT_TYPE = "type";
+
+    @Override
+    public PluginInformation parse(InputStream is) throws IOException {
+        Objects.requireNonNull(is);
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
+            JsonValue json = Json.parse(reader);
+            return parsePluginInformation(json);
+        }
+    }
+
+    private PluginInformation parsePluginInformation(JsonValue json) {
+        JsonObject object = json.asObject();
+
+        String id = parseId(object);
+        String author = parseAuthor(object);
+        ScriptInformation[] scripts = parseScriptInformation(object);
+
+        return new PluginInformation(id, author, scripts);
+    }
+
+    private String parseId(JsonObject json) {
+        JsonValue id = json.get(VALUE_ID);
+        requireValidId(id);
+
+        return id.asString();
+    }
+
+    private void requireValidId(JsonValue id) {
+        requireValueNotNull(id, VALUE_ID);
+        requireValueIsString(id, VALUE_ID);
+    }
+
+    private String parseAuthor(JsonObject json) {
+        JsonValue author = json.get(VALUE_AUTHOR);
+        requireValidAuthor(author);
+
+        return author.asString();
+    }
+
+    private void requireValidAuthor(JsonValue author) {
+        requireValueNotNull(author, VALUE_AUTHOR);
+        requireValueIsString(author, VALUE_AUTHOR);
+    }
+
+    private ScriptInformation[] parseScriptInformation(JsonObject json) {
+        JsonValue value = json.get(VALUE_SCRIPTS);
+        requireValidScripts(value);
+
+        return parseScriptInformation(value.asArray());
+    }
+
+    private void requireValidScripts(JsonValue scripts) {
+        requireValueNotNull(scripts, VALUE_SCRIPTS);
+        requireValueIsArray(scripts, VALUE_SCRIPTS);
+    }
+
+    private ScriptInformation[] parseScriptInformation(JsonArray array) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "parseScriptInformation" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AUnusedPrivateMethod)